src/wp-includes/fonts.php

```diff
@@ -51,3 +51,140 @@ function wp_print_font_faces( $fonts = array() ) {
+ *     Array of information about the upload directory.
+ *
+ *     @type string       $path    Base directory and subdirectory or full path to the fonts upload directory.
+ *     @type string       $url     Base URL and subdirectory or absolute URL to the fonts upload directory.
+ *     @type string       $subdir  Subdirectory
+ *     @type string       $basedir Path without subdir.
+ *     @type string       $baseurl URL path without subdir.
+ *     @type string|false $error   False or error message.
+ * }
+ * @return array $defaults {
+ *     Array of information about the upload directory.
+ *
+ *     @type string       $path    Base directory and subdirectory or full path to the fonts upload directory.
+ *     @type string       $url     Base URL and subdirectory or absolute URL to the fonts upload directory.
+ *     @type string       $subdir  Subdirectory
+ *     @type string       $basedir Path without subdir.
+ *     @type string       $baseurl URL path without subdir.
+ *     @type string|false $error   False or error message.
+ * }
+ */
+function wp_get_font_dir( $defaults = array() ) {
+	// Multi site path
+	$site_path = '';
+	if ( is_multisite() && ! ( is_main_network() && is_main_site() ) ) {
+		$site_path = '/sites/' . get_current_blog_id();
+	}
+
+	// Sets the defaults.
+	$defaults['path']    = path_join( WP_CONTENT_DIR, 'fonts' ) . $site_path;
+	$defaults['url']     = untrailingslashit( content_url( 'fonts' ) ) . $site_path;
+	$defaults['subdir']  = '';
+	$defaults['basedir'] = path_join( WP_CONTENT_DIR, 'fonts' ) . $site_path;
+	$defaults['baseurl'] = untrailingslashit( content_url( 'fonts' ) ) . $site_path;
+	$defaults['error']   = false;
+
+	/**
+	 * Filters the fonts directory data.
+	 *
+	 * This filter allows developers to modify the fonts directory data.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param array $defaults The original fonts directory data.
+	 */
+	return apply_filters( 'font_dir', $defaults );
+}
+
+/**
+ * Deletes child font faces when a font family is deleted.
+ *
+ * @access private
+ * @since 6.5.0
+ *
+ * @param int     $post_id Post ID.
+ * @param WP_Post $post    Post object.
+ */
+function _wp_after_delete_font_family( $post_id, $post ) {
+	if ( 'wp_font_family' !== $post->post_type ) {
+		return;
+	}
+
+	$font_faces = get_children(
+		array(
+			'post_parent' => $post_id,
+			'post_type'   => 'wp_font_face',
+		)
+	);
+
+	foreach ( $font_faces as $font_face ) {
+		wp_delete_post( $font_face->ID, true );
+	}
+}
+
+/**
+ * Deletes associated font files when a font face is deleted.
+ *
+ * @access private
+ * @since 6.5.0
+ *
+ * @param int     $post_id Post ID.
+ * @param WP_Post $post    Post object.
+ */
+function _wp_before_delete_font_face( $post_id, $post ) {
+	if ( 'wp_font_face' !== $post->post_type ) {
+		return;
+	}
+
+	$font_files = get_post_meta( $post_id, '_wp_font_face_file', false );
+
+	foreach ( $font_files as $font_file ) {
+		wp_delete_file( wp_get_font_dir()['path'] . '/' . $font_file );
```

Comment:
`wp_get_font_dir()` is not dependent upon  a specific font file. For performance, it can be moved before the `foreach()` loop, i.e. called once instead of for each loop iteration.

```suggestion
	$font_files = get_post_meta( $post_id, '_wp_font_face_file', false );
	$font_dir   = wp_get_font_dir()['path'];

	foreach ( $font_files as $font_file ) {
		wp_delete_file( $font_dir . '/' . $font_file );
```

Reply:
Addressed in Gutenberg here: https://github.com/WordPress/gutenberg/pull/58691/commits/62487785d063292c9f76ed726687ade8684afc0b

Reply:
Oh awesome, thanks Grant 👍

Reply:
Changed in https://github.com/WordPress/wordpress-develop/pull/6027/commits/095fd857413b24fd674284cd3229e4100cf0d994 ✅